src/django_keycloak/backends.py

```diff
@@ -0,0 +1,81 @@
+"""
+Module containing custom Django authentication backends.
+"""
+from typing import Optional, Union
+from django.contrib.auth.backends import RemoteUserBackend
+from django.contrib.auth import get_user_model
+from django_keycloak.models import KeycloakUserAutoId, KeycloakUser
+from django_keycloak import Token
+
+
+class KeycloakAuthenticationBackend(RemoteUserBackend):
+    """
+    Custom remote backend for Keycloak
+    """
+
+    def authenticate(
+        self,
+        request,
+        remote_user: Optional[str] = None,
+        password: Optional[str] = None,
+    ):
+        """
+        Authenticates an user by credentials, and
+        updates it's information (first name, last name, email).
+        If user does not exist it is created with appropriate permissions.
+
+        Parameters
+        ----------
+        remote_user: str
+            The keycloak username.
```

Comment:
Sorry about this comment earlier. I got confused with another piece of code. I thought `remote_user` was sometimes the username and sometimes the user_id.

I was asking myself why `get_user()` looks up `username` and then `id`. I guess its only there for compatibility reasons?